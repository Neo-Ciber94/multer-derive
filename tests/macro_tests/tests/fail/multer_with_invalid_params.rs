use multer_derive::{Error, FromMultipart, MultipartForm};

#[derive(FromMultipart)]
struct MyStruct {
    #[multer(with = "text_from_multipart")]
    text: Text,
    number: u32,
}

struct Text(String);

fn text_from_multipart(_multipart: &MultipartForm, _ctx: u32) -> Result<Text, Error> {
    todo!()
}

fn main() {}
