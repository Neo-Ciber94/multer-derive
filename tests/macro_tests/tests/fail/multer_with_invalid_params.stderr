error[E0308]: mismatched types
  --> tests/fail/multer_with_invalid_params.rs:3:10
   |
3  | #[derive(FromMultipart)]
   |          ^^^^^^^^^^^^^
   |          |
   |          expected `u32`, found `FormContext<'_>`
   |          arguments to this function are incorrect
   |
note: function defined here
  --> tests/fail/multer_with_invalid_params.rs:12:4
   |
12 | fn text_from_multipart(_multipart: &MultipartForm, _ctx: u32) -> Result<Text, Error> {
   |    ^^^^^^^^^^^^^^^^^^^                             ---------
   = note: this error originates in the derive macro `FromMultipart` (in Nightly builds, run with -Z macro-backtrace for more info)
